snapshots.go

```diff
@@ -231,13 +237,22 @@ func (c *Client) GetSnapshotIsiPath(
 		return "", fmt.Errorf("Snapshot doesn't exist for snapshot id: (%s)", snapshotId)
 	}
 
-	snapshotPath := api.GetRealVolumeSnapshotPathWithIsiPath(isiPath, snapshot.Name)
-	snapshotPath = path.Join(snapshotPath, path.Base(snapshot.Path))
-
-	parts := strings.SplitN(snapshotPath, namespacePath, 2)
-	if len(parts) < 2 {
-		return "", fmt.Errorf("Snapshot doesn't exist for snapshot id: (%s)", snapshotId)
+	//get zone base path
+	zone, err := api.GetZoneByName(ctx, c.API, accessZone)
+	if err != nil {
+		panic(err)
 	}
 
-	return parts[1], nil
+	snapshotPath := api.GetRealVolumeSnapshotPathWithIsiPath(isiPath, zone.Path, snapshot.Name, accessZone)
+	snapshotPath = path.Join(snapshotPath, path.Base(snapshot.Path))
+	//If isi path is different then zone path
```

Comment:
please elaborate this comment

Reply:
@shefali-malhotra hope you have tested with RO volumes from snapshot(source), and not just created RO volumes?